job_list_params.go

```diff
@@ -15,19 +15,36 @@ import (
 // JobListCursor is used to specify a starting point for a paginated
 // job list query.
 type JobListCursor struct {
-	id    int64
-	kind  string
-	queue string
-	time  time.Time
+	id        int64
+	kind      string
+	queue     string
+	sortField JobListOrderByField
+	time      time.Time
 }
 
 // JobListCursorFromJob creates a JobListCursor from a JobRow.
-func JobListCursorFromJob(job *rivertype.JobRow) *JobListCursor {
+func JobListCursorFromJob(job *rivertype.JobRow, sortField JobListOrderByField) *JobListCursor {
+	time := job.CreatedAt
+	switch sortField {
+	case JobListOrderByTime:
+		time = jobListTimeValue(job)
+	case JobListOrderByAttemptedAt:
+		if job.AttemptedAt != nil {
+			time = *job.AttemptedAt
+		}
+	case JobListOrderByFinalizedAt:
+		if job.FinalizedAt != nil {
+			time = *job.FinalizedAt
+		}
+	case JobListOrderByScheduledAt:
+		time = job.ScheduledAt
+	}
```

Comment:
OOC, wasn't this `switch` in `jobListTimeValue` before (and isn't `jobListTimeValue` still in this file?)? Is moving it up just to remove a layer of indirection?

Reply:
Yeah, unfortunately the cursor is tied to the sorting mechanism that was used for the initial query. If the sorting mechanism was the generic 'time' one (meaning - inferred through the state that was queried), we'd have to do the same thing here. If it wasn't, and the user sorted by a specific timestamp field, that needs become a part of the cursor. The logic is slightly different between this function and `jobListTimeValue`, and I couldn't think of a way to combine the functions. Happy to edit if you have thoughts!